docs/docs/hilbert.md

```diff
@@ -0,0 +1,288 @@
+(Hilbert)=
+# The Hilbert module
+
+```{eval-rst}
+.. currentmodule:: netket.hilbert
+```
+
+The [Hilbert](`netket.hilbert`) module defines the abstract Hilbert space API and some concrete implementations, such as {ref}`netket.hilbert.Spin`, {ref}`netket.hilbert.Fock` or others.
+An `Hilbert` object represents a Hilbert space together with a particular choice of computational basis.
+They are needed to construct most other objects in NetKet, but they can also be useful to experiment and validate variational ansätze.
+
+Hilbert space objects are all sub-classes of the abstract class {ref}`netket.hilbert.AbstractHilbert`, which defines the general API respected by all implementations. 
+You can see a birds-eye of the inheritance diagram among the various kinds of Hilbert spaces included with NetKet below (you can click on the nodes in the graph to go to their API documentation page). 
+Classes whose edge is dashed are abstract classes, while the others are concrete and can be instantiated.
+
+```{eval-rst}
+.. inheritance-diagram:: netket.hilbert
+	:top-classes: netket.hilbert.AbstractHilbert
+	:parts: 1
+
+```
+
+{ref}`netket.hilbert.AbstractHilbert` makes very few assumptions on the structure of the resulting space and you will generally very rarely interact with it directly.
+
+There are then two more abstract Hilbert space types: {ref}`netket.hilbert.DiscreteHilbert`, representing Hilbert spaces where the local degrees of freedom are countable, and {ref}`netket.hilbert.ContinuousHilbert`, representing the Hilbert spaces with continuous bases, such as particles in a box. 
+
+Those two abstract types are very different: `ContinuousHilbert` spaces are still experimental and we don't support yet many ways to manipulate them, while `DiscreteHilbert` spaces are much more developed and offer many utilities and handy functionalities.
+
+The most important class of discrete Hilbert spaces are subclasses of {ref}`netket.hilbert.HomogeneousHilbert`, which is a space where the local degrees of freedom are identical among different sites. These subclasses are {ref}`netket.hilbert.Fock`, {ref}`netket.hilbert.Spin`, and {ref}`netket.hilbert.Qubit`.
+
+{ref}`netket.hilbert.TensorHilbert` represents tensor products of different homogeneous hilbert spaces, therefore it is not homogeneous. You can use it to represent composite systems such as spin-boson setups.
+
+{ref}`netket.hilbert.DoubledHilbert` represents a space doubled through [Choi's Isomorphism](https://en.wikipedia.org/wiki/Choi%E2%80%93Jamio%C5%82kowski_isomorphism).
```

Comment:
Somehow that Wikipedia link (which involves trace-preserving complete positive maps) confuses me more: Is `DoubledHilbert` just ℋ ⊗ ℋ or is there more to it?

Reply:
it is.